database/index.js

```diff
@@ -27,4 +27,14 @@ const insertSampleData = (entry) => {
   Reviews.create(entry);
 };
 
-module.exports = { Reviews, insertSampleData };
+const retrieveListing = (id, cb) => {
```

Comment:
Personally I'd have this function return a Promise instead of mixing Promises and the error first callback pattern. I think mixing them might end up confusing/annoying me when I want to call this function. But I don't think it matters much, this works too.